#NAIVE BAYES CLASSIFICATION
**Naive Bayes** is a machine learning classification algorithm that predicts the category of a data point using probabilty. It is called "naive" because it assumes that every feature used for prediction is conditionally independent of every other feature.

In statistics, naive Bayes are simple probabilistic classifiers that apply Bayes’ theorem. This theorem is based on the probability of a hypothesis, given the data and some prior knowledge. The Naive Bayes classifier assumes that all features in the input data are independent of each other, which is often not true in real-world scenarios. However, despite this simplifying assumption, the naive Bayes classifier is widely used because of its efficiency and good performance in many real-world applications.

**Types of Naive Bayes Classification:**
1. Gaussian - continuous data
2. Multinomial - discrete count data
3. Bernoulli - binary data

#Naive Bayes Classification for Diabetes Prediction Using Glucose and Blood Pressure Data

**Dataset Description**


* Sample Size: 995 patient
* 2 Features: Glucose & Blood Pressure
* Target: Diabetes (0/1)


**Methodology:**


*  Data Splitting
*  Model Used: GaussianNB
* Training + Testing


*(Dataset used are extracted from Kaggle)*



In [ ]:
# Importing the Naive bayes classification data from Kaggle

import kagglehub

# Download latest version
path = kagglehub.dataset_download("himanshunakrani/naive-bayes-classification-data")

print("Path to dataset files:", path)

Using Colab cache for faster access to the 'naive-bayes-classification-data' dataset.
Path to dataset files: /kaggle/input/naive-bayes-classification-data


In [ ]:
# Import necessary libraries

import pandas as pd
import os
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, classification_report

In [ ]:
# Load the dataset

csv_file_path = os.path.join(path, 'Naive-Bayes-Classification-Data.csv')
df = pd.read_csv(csv_file_path)
print(df.head())


   glucose  bloodpressure  diabetes
0       40             85         0
1       40             92         0
2       45             63         1
3       45             80         0
4       40             73         1


In [ ]:
# Define the Features and Target

X = df[['glucose', 'bloodpressure']]
y = df['diabetes']

In [ ]:
# Split the dataset into training and testing sets

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [ ]:
# Create a Gaussian Naive Bayes Classifier

gnb = GaussianNB()
gnb.fit(X_train, y_train)

GaussianNB()

In [ ]:
# Make predictions on the testing data

y_pred = model.predict(X_test)

In [ ]:
# Calculate the accuracy of the classifier

accuracy = round(accuracy_score(y_test, y_pred),2)
print("Accuracy:", accuracy)

Accuracy: 0.93


**Result:** The result shows that the Gaussian Naive Bayes model achieved an overall accuracy of 0.93, indicating that the model correctly classified approximately 93% of the test samples. This means that high level of accuracy suggests that glucose and blood pressure are strong predictors of diabetes in the dataset and provide sufficient information for effective classification.


In [ ]:
# Print classification report

print("Classification Report:")
print(classification_report(y_test, y_pred))

Classification Report:
              precision    recall  f1-score   support

           0       0.92      0.92      0.92        93
           1       0.93      0.93      0.93       106

    accuracy                           0.93       199
   macro avg       0.93      0.93      0.93       199
weighted avg       0.93      0.93      0.93       199



**Result:**  The Naive Bayes classifier was used to predict whether respondents are positive or negative for diabetes based on their glucose and blood pressure measurements.

For class 0, the model achieved a precision of 92%, which means that most of the individuals predicted as non-diabetic were correctly identified as true negative cases **(TN)**. It also obtained a recall of 92%, indicating that the model was able to correctly identify most of the non-diabetic while producing only a small number of false positives **(FP)**. The F1-score of 92% further confirms that the model maintains a good balance between precision and recall, demonstrating consistent and reliable performance in classifying non-diabetic cases.

Furthermore, in class 1,  the model achieved a precision of 93%, which means that most of the individuals predicted as diabetic were true positive cases **(TP)**.  It also obtained a recall of 93%, indicating that the model was able to correctly identify most of the actual positive cases while producing only a small number of false negatives **(FN)**. The F1-score of 93% further confirms that the model has a good balance between precision and recall, showing consistent and reliable performance in classification.


**References:**

DataTechNotes. (2024, March 25). Gaussian Naive Bayes Classification Example with Scikit-learn [Video]. YouTube. https://www.youtube.com/watch?v=wum7ZTC8iiw

GeeksforGeeks. (2026, February 27). Naive Bayes classifiers. GeeksforGeeks. https://www.geeksforgeeks.org/machine-learning/naive-bayes-classifiers/

Kashyap, P. (2024, December 2). Understanding Precision, Recall, and F1 Score Metrics. Medium. https://medium.com/@piyushkashyap045/understanding-precision-recall-and-f1-score-metrics-ea219b908093

Ray, S. (2025, May 1). Naive Bayes classifier explained with practical problems. Analytics Vidhya. https://www.analyticsvidhya.com/blog/2017/09/naive-bayes-explained/#:~:text=The%20Naive%20Bayes%20classifier%20is,tasks%20such%20as%20text%20classification.